projects/etos_suite_runner/src/etos_suite_runner/lib/esr_parameters.py

```diff
@@ -151,7 +152,17 @@ def tercc(self) -> Union[list[dict], dict]:
         :return: Test execution event.
         """
         if self.etos.config.get("tercc") is None:
-            tercc = json.loads(os.getenv("TERCC", "{}"))
+            tercc = EiffelTestExecutionRecipeCollectionCreatedEvent()
+            tercc_json = None
+            if os.getenv("TERCC") is not None:
+                # first option for backwards compatibility
+                self.logger.info("Reading TERCC from environment variable")
+                tercc_json = json.loads(os.getenv("TERCC", "{}"))
+                tercc.rebuild(tercc_json)
+            else:
+                # requires testrun custom resource defined in Kubernetes
+                self.logger.info("Reading TERCC from Kubernetes testrun resource")
+                tercc = TestRun(Kubernetes()).get(os.getenv("TESTRUN"))
```

Comment:
A few problems with this.

1. This property should only be used as a fallback for when we're not running in the controller environment (line 123).
2. The TestRun resource from Kubernetes is not equivalent to a TERCC.
3. The code for requesting a TestRun resource from Kubernetes instead of loading the TERCC environment variable as a TestRun should be in the `test_suite` property (line 174)